src/components/Tank/Tank.model.ts

```diff
@@ -0,0 +1,14 @@
+export declare type TankPosition = 'up' | 'down' | 'left' | 'right';
+export declare type TankColor = 'yellow' | 'white' | 'green' | 'red';
+export declare type TankType = '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7';
+
+export interface TankProps {
```

Comment:
- [ ] With the new Context-based technique, we're now passing state instead of props(take a look at the Gerb component). That means you can rename this to TankState and rename the file to tank.state.ts as all the other types are related to the state.